geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/rebalance/model/Member.java

```diff
@@ -55,6 +55,31 @@
     this.localMaxMemory = localMaxMemory;
   }
 
+  /**
+   * @param sourceMember the member we will be moving this bucket off of
+   * @param checkZone true if we should not put two copies of a bucket on two nodes with the same
+   *        IP address.
+   */
+  public RefusalReason canDelete(Bucket bucket, InternalDistributedMember sourceMember,
+      boolean checkZone) {
```

Comment:
I'm not sure the logic in this method is correct. What if we have an extra copy of the bucket, and each copy is in a different redundancy zone? It seems like this would prevent us from deleting the extra copy when if fact we should.

Maybe we could keep this method but only throw an exception if we find two copies of the bucket in the same zone *and* we are trying to delete a copy in a different zone, or something like that.